include/boost/phoenix/core/actor.hpp

```diff
@@ -143,6 +143,11 @@ namespace boost { namespace phoenix
             >::type
             expr_type;
 
+#ifndef BOOST_NO_CXX11_DEFAULTED_FUNCTIONS
+        actor() = default;
+        actor(const actor&) = default;
```

Comment:
Doing this disables implicit move constructor/assignment, is not it?

Reply:
BOOST_PROTO_EXTENDS_ASSIGN defines a copy assignment (which is why I get this error initially in gcc 9). So I think that today there is no implicit move constructor/assignments.

Reply:
True. This should be a bug...

Reply:
To be honest, I am not sure how to deal with this. Is there a way to make a minimal test code so we can ask the proto maintainer?

Reply:
What you want to ask the proto maintainer about? The only thing on the proto side is why `BOOST_PROTO_EXTENDS_ASSIGN` macro adds a copy assignment operator. The Phoenix probably needs only a template assignment operator.

Reply:
> why `BOOST_PROTO_EXTENDS_ASSIGN` macro adds a copy assignment operator

That one ^. Isn't that the issue here that we are trying to fix in the client(phoenix) side?

Reply:
It is the root cause of the warning, yes.

Reply:
> It is the root cause of the warning, yes.

So we're probably "fixing" the wrong side.

Reply:
It depends on where the self-assignment of Phoenix placeholders is allowed or not.

Reply:
And I think it cannot be disallowed because the signature of two actors could be the same, but the content is different.

Reply:
It turns out that I have changed the semantics in https://github.com/boostorg/phoenix/commit/1e66ae741f10d185863d7d8d8a54d562efe5144e#diff-85e553336a9d92ba0afb347581e0f32eL207-L212. By removing the copy assignment operator that comes from `BOOST_PROTO_EXTENDS_ASSIGN` we will revert to the previous behavior, but in this case we will again break:
```cpp
#include <boost/phoenix.hpp>
#include <iostream>

int main()
{
  using namespace boost::phoenix;

  int i = 0, j = 123;
  auto ri = ref(i); 
  (ri = ref(j))();
  std::cout << "i=" << i << ", j=" << j << "\n";
}
```
I did not find a ticket that it was broken before, and I checked that the test suit does not cover this usage, but it looks reasonable not break it again.

Reply:
So I am not sure I am following what shall we do now to silence this warning.

Reply:
So, I don't follow here. Is there still a problem to be solved now on the development branch ? If not, I will just close this pull request.